pkg/chunked/compression_linux.go

```diff
@@ -175,26 +186,29 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 	if tarSplitChunk.Offset > 0 {
 		chunks = append(chunks, tarSplitChunk)
 	}
-	parts, errs, err := blobStream.GetBlobAt(chunks)
+
+	streamsOrErrors, err := getBlobAt(blobStream, chunks, len(chunks))
 	if err != nil {
 		return nil, nil, nil, 0, err
 	}
 
+	defer func() {
```

Comment:
Having to consume everything there is, generally, an undesired cost.

The typical handling for that is to use a cancellable context, and to have the consumer here `defer cancelTheContext`; and for the goroutine to observe that context and terminate on cancellation.

Sadly that is not possible in `ImageSourceSeekable` (no `Context` parameter), but maybe we could build the `streamsOrErrors` the idiomatic way, and have `getBlobAt` deal with draining the produced output, instead of doing that in every consumer.

(I didn’t try, this might be too hard to do. Notably channel sends in `getBlobAt` would also have to `select` vs. the context terminating.)